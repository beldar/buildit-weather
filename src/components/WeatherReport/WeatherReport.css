.weather-report {
  border: 1px solid #cbcbcb;
}

.weather-report__days {
  display: flex;
  flex-direction: row;
}

@media all and (max-width: 980px) {
  .weather-report__days {
    flex-direction: column;
  }
}
