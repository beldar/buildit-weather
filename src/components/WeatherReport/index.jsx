import React from 'react';
import WeatherDay from '../WeatherDay';
import './WeatherReport.css';

export const groupByDay = function(list) {
  return list.reduce((obj, cur) => {
    let date = cur.dt_txt.split(' ')[0];
    obj[date] = obj[date] || [];
    obj[date].push(cur);
    return obj;
  }, {});
};

export default function WeatherReport(props) {
  if (!props || !props.list || !props.list.length) return null;

  const days = groupByDay(props.list);

  return (
    <div className="weather-report">
      <ul className="weather-report__days">
        { Object.values(days).map((day, k) => <WeatherDay key={k} day={ day } />) }
      </ul>
    </div>
  )
};
