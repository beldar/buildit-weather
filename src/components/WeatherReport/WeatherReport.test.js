import React from 'react';
import WeatherReport, { groupByDay } from './index';
import renderer from 'react-test-renderer';

describe('WeatherReport', () => {
  it('should not render if list prop is not passed', () => {
    const tree = renderer.create(
      <WeatherReport list={null} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly if list prop is passed', () => {
    const list = [{
      dt: '1499342400',
      dt_txt: '2017-07-06 12:00:00',
      main: {
        temp: 22.5,
        humidity: 60
      },
      weather: [{
        description: 'cloudy'
      }]
    }];

    const tree = renderer.create(
      <WeatherReport list={list} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('WeatherReport/groupByDay', () => {
    it('should group data by day', () => {
      const input = [
        { dt_txt: '2017-07-03 15:00:00'},
        { dt_txt: '2017-07-03 16:00:00'},
        { dt_txt: '2017-07-04 15:00:00'},
        { dt_txt: '2017-07-04 15:00:00'}
      ];
      const output = {
        '2017-07-03': [
          { dt_txt: '2017-07-03 15:00:00'},
          { dt_txt: '2017-07-03 16:00:00'}
        ],
        '2017-07-04': [
          { dt_txt: '2017-07-04 15:00:00'},
          { dt_txt: '2017-07-04 15:00:00'}
        ]
      };

      expect(groupByDay(input)).toEqual(output);
    })
  })
});
