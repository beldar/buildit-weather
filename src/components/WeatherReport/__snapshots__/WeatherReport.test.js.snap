// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherReport should not render if list prop is not passed 1`] = `null`;

exports[`WeatherReport should render correctly if list prop is passed 1`] = `
<div
  className="weather-report"
>
  <ul
    className="weather-report__days"
  >
    <li
      className="weather-day"
    >
      <h2
        className="weather-day__title"
      >
        Thursday 6th of July
      </h2>
      <ul>
        <li
          className="weather-hour-item"
        >
          <ul
            className="weather-hour"
          >
            <li
              className="weather-hour__hour"
            >
              ⏱ 
              12:00:00
            </li>
            <li>
              <span
                className="weather-hour__prop"
              >
                Temperature:
              </span>
               
              <strong>
                22.5
                ºC
              </strong>
            </li>
            <li>
              <span
                className="weather-hour__prop"
              >
                Humidity:
              </span>
               
              <strong>
                60
                %
              </strong>
            </li>
            <li>
              <span
                className="weather-hour__prop"
              >
                Weather: 
              </span>
               
              <strong>
                cloudy
              </strong>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
